src/client/http_client.rs

```diff
@@ -790,8 +793,21 @@ impl SearchQueryRequest {
 
     pub fn payload(&self) -> Option<Vec<u8>> {
         match self {
-            Self::Execute { query, timeout, .. } => {
-                let json = json!({ "query": { "query": query }, "ctl": { "timeout": timeout }});
+            Self::Execute {
+                query,
+                timeout,
+                vector,
```

Comment:
I also feel like a new VectorQueryRequest could be nice just to make intent clear and better code clarity, so that we can simplify the if below..